util/locking.py

```diff
@@ -26,13 +26,26 @@ class GlobalLock(object):
     critical code paths.
     """
 
+    lock_factory = None
+
+    def __new__(cls):
+        if cls.lock_factory is None:
+            cls.lock_factory = RedLockFactory(
+                connection_details=[
+                    {
+                        **dict(app.config["USER_EVENTS_REDIS"]),
```

Comment:
Is this only for the Redis instance being used for user events? Technically there can be separate Redis instances for user events and the build manager.

Reply:
Yes, `USER_EVENTS_REDIS` was the one used by default prior to this change. Do we want to make this configurable? Especially if we intend to use this elsewhere (#444)

Reply:
We can add this later.